lib/elixir/src/elixir_rewrite.erl

```diff
@@ -20,218 +21,245 @@
-  {maps, find, [Key, Map]};
-rewrite(?map, 'fetch!', [Map, Key]) ->
-  {maps, get, [Key, Map]};
-rewrite(?map, 'has_key?', [Map, Key]) ->
-  {maps, is_key, [Key, Map]};
-rewrite(?map, put, [Map, Key, Value]) ->
-  {maps, put, [Key, Value, Map]};
-rewrite(?map, 'replace!', [Map, Key, Value]) ->
-  {maps, update, [Key, Value, Map]};
-rewrite(?port, monitor, [Arg]) ->
-  {erlang, monitor, [port, Arg]};
-rewrite(?process, group_leader, [Pid, Leader]) ->
-  {erlang, group_leader, [Leader, Pid]};
-rewrite(?process, monitor, [Arg]) ->
-  {erlang, monitor, [process, Arg]};
-rewrite(?process, send_after, [Dest, Msg, Time]) ->
-  {erlang, send_after, [Time, Dest, Msg]};
-rewrite(?process, send_after, [Dest, Msg, Time, Opts]) ->
-  {erlang, send_after, [Time, Dest, Msg, Opts]};
-rewrite(?string, to_atom, [Arg]) ->
-  {erlang, binary_to_atom, [Arg, utf8]};
-rewrite(?string, to_existing_atom, [Arg]) ->
-  {erlang, binary_to_existing_atom, [Arg, utf8]};
-rewrite(?tuple, delete_at, [Tuple, Index]) ->
+?rewrite(?atom, to_string, [Arg], erlang, atom_to_binary, [Arg, utf8]);
+?rewrite(?kernel, is_map_key, [Map, Key], erlang, is_map_key, [Key, Map]);
+?rewrite(?map, delete, [Map, Key], maps, remove, [Key, Map]);
+?rewrite(?map, fetch, [Map, Key], maps, find, [Key, Map]);
+?rewrite(?map, 'fetch!', [Map, Key], maps, get, [Key, Map]);
+?rewrite(?map, 'has_key?', [Map, Key], maps, is_key, [Key, Map]);
+?rewrite(?map, put, [Map, Key, Value], maps, put, [Key, Value, Map]);
+?rewrite(?map, 'replace!', [Map, Key, Value], maps, update, [Key, Value, Map]);
+?rewrite(?port, monitor, [Arg], erlang, monitor, [port, Arg]);
+?rewrite(?process, group_leader, [Pid, Leader], erlang, group_leader, [Leader, Pid]);
+?rewrite(?process, monitor, [Arg], erlang, monitor, [process, Arg]);
+?rewrite(?process, send_after, [Dest, Msg, Time], erlang, send_after, [Time, Dest, Msg]);
+?rewrite(?process, send_after, [Dest, Msg, Time, Opts], erlang, send_after, [Time, Dest, Msg, Opts]);
+?rewrite(?string, to_atom, [Arg], erlang, binary_to_atom, [Arg, utf8]);
+?rewrite(?string, to_existing_atom, [Arg], erlang, binary_to_existing_atom, [Arg, utf8]);
+?rewrite(?tuple, duplicate, [Data, Size], erlang, make_tuple, [Size, Data]);
+
+inner_rewrite(ex_to_erl, ?tuple, delete_at, [Tuple, Index]) ->
   {erlang, delete_element, [increment(Index), Tuple]};
-rewrite(?tuple, duplicate, [Data, Size]) ->
-  {erlang, make_tuple, [Size, Data]};
-rewrite(?tuple, insert_at, [Tuple, Index, Term]) ->
+inner_rewrite(ex_to_erl, ?tuple, insert_at, [Tuple, Index, Term]) ->
   {erlang, insert_element, [increment(Index), Tuple, Term]};
-rewrite(Receiver, Fun, Args) ->
-  {Receiver, Fun, Args}.
+inner_rewrite(ex_to_erl, ?kernel, elem, [Tuple, Index]) ->
+  {erlang, element, [increment(Index), Tuple]};
+inner_rewrite(ex_to_erl, ?kernel, put_elem, [Tuple, Index, Value]) ->
+  {erlang, setelement, [increment(Index), Tuple, Value]};
+
+inner_rewrite(erl_to_erl, erlang, delete_element, [Index, Tuple]) when is_number(Index) ->
```

Comment:
`erl_to_erl` are typos, right?

Reply:
If so, we are probably missing some tests. :)

Reply:
Advice on where to put the tests?

Reply:
I would ideally check it on the module check error messages to see if they were properly reversed. Thoughts?